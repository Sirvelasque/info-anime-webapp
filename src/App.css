html {
  font-family: 'Lato', sans-serif;
}

body {
  color: white;
}

.Loading {
  position: fixed;
  top: 20%;
  height: 400px;
  width: 100%;
  background: linear-gradient(#973158b7, rgba(0, 0, 0, 0.712));
  font-size: 48px;
  line-height: 54px;
  text-align: center;
  padding-top: 170px;
}
