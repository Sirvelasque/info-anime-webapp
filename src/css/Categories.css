.categoriesChard h2 {
  margin: 20px 8px;
}

.categoriesChard h4 {
  margin: 20px;
}

.animeitem a .chardTitle {
  display: flex;
  flex-wrap: wrap;
  margin-right: 10px;
}

.animeitem a {
  text-decoration: none;
  color: white;
  display: flex;
}

.animeitem {
  display: flex;
  flex-wrap: nowrap;
  background: linear-gradient(#e54986, black);
  margin-bottom: 30px;
}

.fa-heart {
  width: 100%;
  height: 20px;
  margin-top: 20px;
}

.wrap {
  display: flex;
}

.description {
  display: flex;
  align-items: center;
}
