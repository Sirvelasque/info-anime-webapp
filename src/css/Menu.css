.categories {
  margin-top: 30px;
  display: flex;
  justify-self: center;
  flex-wrap: wrap;
  width: 100%;
  justify-content: center;
}

body {
  color: white;
}

.categories a {
  width: 100%;
}

.genre {
  width: 50%;
  height: 43vh;
  border: none;
  color: white;
  font-size: 42px;
  line-height: 48px;
  padding-top: 370px;
}

.Action {
  background:
    linear-gradient(black, #e54986),
    url("../resource/action.png");
  background-repeat: no-repeat;
  background-size: 120%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

.Comedy {
  background:
    linear-gradient(black, #d04278),
    url("../resource/comedy.jpg");
  background-repeat: no-repeat;
  background-size: 128.3%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

.Gourmet {
  background:
    linear-gradient(black, #e54986),
    url("../resource/gourmet.jpg");
  background-repeat: no-repeat;
  background-size: 120%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

.Fantasy {
  background:
    linear-gradient(black, #d04278),
    url("../resource/fantasy.jpg");
  background-repeat: no-repeat;
  background-size: 120%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

.Drama {
  background:
    linear-gradient(black, #e54986),
    url("../resource/drama.jpg");
  background-repeat: no-repeat;
  background-size: 120%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

.Slice {
  background:
    linear-gradient(black, #d04278),
    url("../resource/slice.jpg");
  background-repeat: no-repeat;
  background-size: 144.4%;
  background-blend-mode: exclusion;
  cursor: pointer;
}

@media screen and (max-width: 980px) {
  .genre {
    height: 200px;
    padding-top: 100px;
  }
}
