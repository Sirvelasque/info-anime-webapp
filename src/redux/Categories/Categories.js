const CATEGORIECHANGE = 'CATEGORIECHANGE';

const categorieReducer = (state = 'Action', action) => {
  switch (action.type) {
    case CATEGORIECHANGE:
      return action.payload;
    default:
      return state;
  }
};

const getCategorie = (cat) => async (dispatch) => {
  dispatch({
    type: CATEGORIECHANGE,
    payload: cat,
  });
};

export default categorieReducer;
export { getCategorie };
