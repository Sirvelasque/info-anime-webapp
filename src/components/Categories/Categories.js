import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getAnimes } from '../../redux/Companies/Companies';
import Categorie from './categorie';

import '../../css/Categories.css';

const chard = (anime) => {
  if (anime.length !== 0) {
    return (
      <Categorie
        id={anime.data.data.mal_id}
        name={anime.data.data.title}
        img={anime.data.data.images.webp.image_url}
        description={anime.data.data.synopsis}
        likes={anime.data.data.score}
      />
    );
  }
  return null;
};

const Categories = () => {
  const data = useSelector((state) => state);
  const dispatch = useDispatch();
  useEffect(() => {
    if (data.animes.length === 0) {
      dispatch(getAnimes());
    }
  }, []);
  const animeList = data.animes.filter((e) => {
    for (let i = 0; i < e.data.data.genres.length; i += 1) {
      if (e.data.data.genres[i].name === data.categorie) return true;
    }
    return false;
  });
  return (
    <div>
      <div className="categoriesChard">
        <h2>{data.categorie}</h2>
        {animeList.map((e) => chard(e))}
      </div>
    </div>
  );
};

export default Categories;
