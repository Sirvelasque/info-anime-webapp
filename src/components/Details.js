import { useSelector } from 'react-redux';

import '../css/details.css';

const Details = () => {
  const data = useSelector((state) => state.details.data.data);

  return (
    <div>
      <div className="Container">
        <div className="infoContainer">
          <div className="Head">
            <h2>{data.title}</h2>
          </div>
          <div className="Numbers">
            <ul className="list">
              <li className="fli">
                Popularity:
                {data.popularity}
              </li>
              <li className="sli">
                Rank:
                {data.rank}
              </li>
              <li className="fli">
                Score:
                {data.score}
                <br />
                Scored by:
                {data.scored_by}
              </li>
              <li className="sli">
                Year:
                {data.year}
              </li>
            </ul>
          </div>
        </div>
        <div className="synopsis">
          <img alt="Anime one" src={data.images.jpg.large_image_url} />
          <h4>Synopsis</h4>
          <hr />
          <div className="text">
            {data.synopsis}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Details;
