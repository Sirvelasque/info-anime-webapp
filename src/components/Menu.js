import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { getAnimes } from '../redux/Companies/Companies';
import { getDetails } from '../redux/details/details';
import { getCategorie } from '../redux/Categories/Categories';

import '../css/Menu.css';

let init = 0;

const Menu = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    if (init === 0) {
      dispatch(getAnimes());
      dispatch(getDetails(1));
      init = 2;
    }
  }, [dispatch]);

  const sendCategorie = async (cat) => {
    dispatch(getCategorie(cat));
  };

  return (
    <div className="categories">
      <h2>Categories</h2>
      <NavLink to="Categories">
        <button type="button" className="genre Action" onClick={() => sendCategorie('Action')}>Action</button>
        <button type="button" className="genre Comedy" onClick={() => sendCategorie('Comedy')}>Comedy</button>
        <button type="button" className="genre Fantasy" onClick={() => sendCategorie('Fantasy')}>Fantasy</button>
        <button type="button" className="genre Gourmet" onClick={() => sendCategorie('Gourmet')}>Gourmet</button>
        <button type="button" className="genre Drama" onClick={() => sendCategorie('Drama')}>Drama</button>
        <button type="button" className="genre Slice" onClick={() => sendCategorie('Slice of life')}>Slice of life</button>
      </NavLink>
    </div>
  );
};

export default Menu;
