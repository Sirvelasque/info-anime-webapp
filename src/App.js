import { BrowserRouter, Route, Routes } from 'react-router-dom';
import NavBar from './components/NavBar';
import Menu from './components/Menu';
import Categories from './components/Categories/Categories';
import Details from './components/Details';
import './App.css';

function App() {
  return (
    <BrowserRouter>
      <NavBar />
      <Routes>
        <Route path="/" element={<Menu />} />
        <Route path="categories" element={<Categories />} />
        <Route path="details" element={<Details />} />
      </Routes>
    </BrowserRouter>
  );
}

export default App;
