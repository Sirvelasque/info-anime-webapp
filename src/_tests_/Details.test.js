import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Details from '../components/Details';
import Menu from '../components/Menu';
import Navbar from '../components/NavBar';
import Categories from '../components/Categories/Categories';
import App from '../App'

test('render details', () => {
  const Container = () => {
    <BrowserRouter>
      <Details />
    </BrowserRouter>;
  };
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Menu', () => {
  const Container = () => {
    <BrowserRouter>
      <Menu />
    </BrowserRouter>;
  };
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Categories', () => {
  const Container = () => {
    <BrowserRouter>
      <Categories />
    </BrowserRouter>;
  };
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Navbar', () => {
  const Container = () => {
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>;
  };
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render App', () => {
  const Container = () => {
    <BrowserRouter>
      <App />
    </BrowserRouter>;
  };
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});
