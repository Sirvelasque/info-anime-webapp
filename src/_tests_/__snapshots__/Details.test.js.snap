// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `null`;

exports[`Render Categories 1`] = `null`;

exports[`Render Navbar 1`] = `null`;

exports[`render details 1`] = `null`;

exports[`renders Menu 1`] = `null`;
